### ISSUE TITLE
IfcOpenShell-Python: Proper Conda / PyPI mirror

### ISSUE THREAD

**Issue by MaximilianFranz** [IfcOpenShell/IfcOpenShell#935]
There seems to be a breaking difference in the builds that are available from ifcopenshell.org and those on conda-forge. Any plans on having a trigger that pushes newest releases to conda as well? 

Because for managed environment and build pipelines, the variant downloading the package from ifcopenshell.org is not really clean. 

How can we help out there? 

Thanks in advance!

**Comment by aothms** [comment]
> There seems to be a breaking difference in the builds that are available from ifcopenshell.org and those on conda-forge.

Do you have details?

> Any plans on having a trigger that pushes newest releases to conda as well?

See the discussion here. https://github.com/conda-forge/ifcopenshell-feedstock/pull/8#issuecomment-600689777 I'm fine with everything, just tell me what to tag.

> the variant downloading the package from ifcopenshell.org is not really clean.

Do you have details? I still expect the statically linked version to be a bit faster, but I didn't benchmark it.

**Comment by MaximilianFranz** [comment]
I was getting an error when parsing a shape: 
```python 
>       return _ifcopenshell_wrapper.create_shape(settings, instance, representation)
E       RuntimeError: Failed to process shape
``` 
which I don't get if I `wget` ifcopenshell from the website and unzip / link it manually into the site-packages of my conda environment. 

The last point was merely referencing the ease of setup. If I can just put ifcopenshell in my `environment.yaml`, that would be the best setup. Currently, we have a setup_env.sh runnning that installs the conda environment and then manually unpacks the ifcopenshell version from the website into the site-packages. 

I haven't looked into the details, what version / commit is fixing the issue we have. Neither am I very fond of the details when it comes to statically linked / dynamically linked versions. Can you point out the differences there with regard to ifcopenshell?

**Comment by aothms** [comment]
> Failed to process shape

Yes these things can be very specifically related to one commit and the version on conda is indeed a couple months (?) old.

> environment.yaml ... manually unpacks 

Ok, I see, that indeed makes a difference. 

> Can you point out the differences there with regard to ifcopenshell?

It's simply that with dynamic linking you have DLLs that export functions. E.g. to get a X-coordinate of a point you call the function in OCCT gp_Pnt::X(). A function call for simply returning a memory address is expensive. Therefore most compilers/linkers will typically inline this and optimize this to simply reading a memory address and further optimization steps can make use of this knowledge. I never really benchmarked this though.

Well. Like I said see/follow up on the discussion here: https://github.com/conda-forge/ifcopenshell-feedstock/pull/8#issuecomment-600689777

**Comment by Moult** [comment]
Closing as resolved :) See #2132 #2293
